app/src/main/java/com/fpvout/digiview/dvr/DVR.java

```diff
@@ -0,0 +1,135 @@
+package com.fpvout.digiview.dvr;
+
+import android.app.Activity;
+import android.media.MediaRecorder;
+import android.os.Environment;
+import android.os.Handler;
+import android.util.Log;
+import android.widget.ImageButton;
+import android.widget.Toast;
+
+import com.fpvout.digiview.R;
+import com.fpvout.digiview.VideoReaderExoplayer;
+import com.fpvout.digiview.helpers.StreamDumper;
+
+import java.io.File;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+
+public class DVR {
+    private static final int WRITE_EXTERNAL_STORAGE = 0;
+    private final Activity activity;
+    private boolean recordAmbientAudio;
+    private MediaRecorder recorder;
+    private boolean recording = false;
+    private static DVR instance;
+    private static final String DVR_LOG_TAG = "DVR";
+    private String defaultFolder = "";
+    private String ambietAudio;
+    private String videoFile;
+    private String dvrFile;
+    private VideoReaderExoplayer mPlayer;
+    private File dvrTmpFile;
+    private String fileName;
+    private StreamDumper streamDumper;
+    private static Handler updateAfterRecord;
+    public static final String LATEST_THUMB_FILE = "latest.jpeg";
+
+    DVR(Activity activity, boolean recordAmbientAudio, Handler updateAfterRecord){
+        this.activity = activity;
+        this.recordAmbientAudio = recordAmbientAudio;
+        this.updateAfterRecord = updateAfterRecord;
+        defaultFolder =  Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_PICTURES) + "/" + this.activity.getApplicationInfo().loadLabel(this.activity.getPackageManager()).toString();
+        streamDumper = new StreamDumper(activity, defaultFolder);
+    }
+
+    public static DVR getInstance(Activity context, boolean recordAmbientAudio, Handler updateAfterRecord){
+        if (instance == null) {
+            instance = new DVR(context, recordAmbientAudio, updateAfterRecord);
+        }
+        return instance;
+    }
+
+    public void init(VideoReaderExoplayer mPlayer) throws IOException {
+        this.mPlayer = mPlayer;
```

Comment:
ideally, we don't want DVR to depend on video Player (they should be different features)

Reply:
should be easy to remove that cause it just used to determine if the stream is running ... I will implement a different solution.